Add support for running the function app from the azure organisation search deployment pipeline

neilbmclaughlin opened the pull request.

st3v3nhunt commented on GetIndexDefinition/index.js:
```diff
@@ -0,0 +1,13 @@
+    delete indexDefinition.name;
+    return indexDefinition;
+  }
+  throw Error(`Could not get index definition for ${indexName}`);
```
It might be worth adding the statusCode and/or, if there is an error message, adding that to the error to aid with debugging.

st3v3nhunt commented on GetIndexNames/index.js:
```diff
@@ -2,9 +2,11 @@ const azureApimRequest = require('../lib/AzureApimRequest');
 const utilities = require('../lib/utilities');
 
 module.exports = async function getIndexNames(context, apimApiName) {
+  context.log({ apimApiName });
```
Could some additional information be added to increase the usefulness of this message within the logs please?

st3v3nhunt commented on GetIndexNames/index.js:
```diff
@@ -2,9 +2,11 @@ const azureApimRequest = require('../lib/AzureApimRequest');
 
   const activeServiceUrl = response.properties.serviceUrl;
+  context.log({ activeServiceUrl });
 
```
Could some additional information be added to increase the usefulness of this message within the logs please?

st3v3nhunt commented on GetIndexNamesOrchestrator/index.js:
```diff
@@ -0,0 +1,14 @@
+  context.log('Starting GetIndexNamesOrchestrator');
+  const apimApiName = context.df.getInput();
+
+  context.log({ apimApiName });
```
Could some additional information be added to increase the usefulness of this message within the logs please?

st3v3nhunt commented on GetIndexNamesOrchestrator/index.js:
```diff
@@ -0,0 +1,14 @@
+
+  const indexNames = yield context.df.callActivity('GetIndexNames', apimApiName);
+
+  context.log({ indexNames });
```
Could some additional information be added to increase the usefulness of this message within the logs please?

st3v3nhunt commented on OrchestratorFunction/index.js:
```diff
@@ -1,42 +1,31 @@
+    indexDefinition,
+    indexNames,
+  };
+  const returnStatus = yield context.df.callSubOrchestrator(
```
Possibly return immediately rather than assigning to a variable?

neilbmclaughlin replied:
Added the missing logging which was the reason for having the holding variable.

st3v3nhunt commented on ReIndex/index.js:
```diff
@@ -8,14 +8,9 @@ async function azureIndexersSearchRequest(indexerName, method, body) {
-    await azureIndexesSearchRequest(targetIndexName, 'put', JSON.stringify(sourceIndexDefinition));
-  }
+async function recreateIndex(indexName, indexDefinition) {
+  await azureIndexesSearchRequest(indexName, 'delete');
```
Should or can the statusCode of the response be checked to ensure it is ok rather than just assuming it was ok?

st3v3nhunt commented on ReIndex/index.js:
```diff
@@ -8,14 +8,9 @@ async function azureIndexersSearchRequest(indexerName, method, body) {
-  }
+async function recreateIndex(indexName, indexDefinition) {
+  await azureIndexesSearchRequest(indexName, 'delete');
+  await azureIndexesSearchRequest(indexName, 'put', JSON.stringify(indexDefinition));
```
Same as above, should the response be checked?

st3v3nhunt commented on ReIndexOrchestrator/index.js:
```diff
@@ -0,0 +1,44 @@
+  context.log('Starting ReIndexOrchestrator');
+  const input = context.df.getInput();
+
+  context.log({ input });
```
Could some additional information be added to increase the usefulness of this message within the logs please?

st3v3nhunt commented on ReIndexOrchestrator/index.js:
```diff
@@ -0,0 +1,44 @@
+  if (!indexDefinition) {
+    indexDefinition = yield context.df.callActivity('GetIndexDefinition', indexNames.active);
+  }
+  context.log(indexDefinition);
```
Other log messages dealing with variables log json objects rather than just the variable. Should this also use that convention?

st3v3nhunt commented on ReIndexOrchestrator/index.js:
```diff
@@ -0,0 +1,44 @@
+  context.log(indexDefinition);
+
+  const indexerName = yield context.df.callActivity('ReIndex', { indexDefinition, indexNames });
+  context.log({ indexerName });
```
Could some additional information be added to increase the usefulness of this message within the logs please?

st3v3nhunt commented on ReIndexOrchestrator/index.js:
```diff
@@ -0,0 +1,44 @@
+      .add(polling.interval, polling.units);
+    yield context.df.createTimer(nextCheck.toDate());
+    const indexerStatus = yield context.df.callActivity('GetIndexerStatus', indexerName);
+    context.log({ indexerStatus });
```
Could some additional information be added to increase the usefulness of this message within the logs please?

neilbmclaughlin commented:
I've updated the readme to explain the `body*.json` files and moved them to their own directory.

I'm going to leave the logging as a separate piece of work and have raised an issue for this (#12).

Regarding the commented console.log lines they can be removed as part of this work too.
